### PR TITLE
Update README for the new UUID & __init__ syntax

### PR DESCRIPTION
**Pull request by timholy**
There still seems to be some confusion about how to use this (https://github.com/MikeInnes/Requires.jl/pull/46#issuecomment-415725554), so let's document the new syntax in the README. CC @davidavdav

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,34 +11,85 @@ faster, maybe. It supports specifying glue code in packages which will
...
+```julia
+function __init__()
+    @require Gadfly="c91e804a-d5a3-530f-b6f0-dfbca275c004" @eval using MyGluePkg
 end
 ```
 
-For larger amounts of code you can also use `@require Package include("glue.jl")`.
-The code wrapped by `@require` will execute as soon as the given package is loaded
-(which may be immediately).
+if you wish to exploit precompilation for the new code.
```
Comment (tlnagy) [inline README.md]: So unless you package your glue code in a module it won't be precompiled?

Comment (timholy) [reply README.md]: Correct.

Comment (DilumAluthge) [reply README.md]: @timholy Does MyGluePkg need to be a separate Julia package with a separate repository?  Or is there a way for the MyGluePkg code to be stored in the same repo as MyPkg?

Comment (timholy) [reply README.md]: There is no official support for that, but `MyPkg` could do a `push!(LOAD_PATH, @__DIR__)` and make itself its own little package repository.

Comment (DilumAluthge) [reply README.md]: Would that be an acceptable approach for package authors to take, or is that considered bad practice?

**Comment by davidavdav** [comment]
Thanks for updating the README.  What about julia Base packages, such as `LinearAlgebra`, that don't seem to have a UUID?

File: README.md (hunk 1 above)
Comment (timholy) [reply README.md]: I don't have any kind of firm opinion on that, sorry.

**Comment by timholy** [comment]
They do, though I'm not sure where to find them officially. They do appear in your `~/.julia/environments/vX.X/Manifest.toml` file.

**Comment by timholy** [comment]
I'm not sure you need to trigger on them, however, since they are always present. Just make your package dependent on them.

**Comment by davidavdav** [comment]
OK, thanks, yes, that is how it was before.  I just thought it'd be cleaner to separate the LinearAlgebra support.

**Comment by MikeInnes** [comment]
Thanks a lot @timholy!

**Pull request merged**